PennyMe/MachineChangedRequest.swift

```diff
@@ -0,0 +1,326 @@
+                        secondaryButton: .cancel(Text("No")) {
+                            showDoneAlert = false
+                            
+                        }
+                    )
+                }
+            }
+        }
+    }
+}
+    
+
+@available(iOS 14.0, *)
+struct MachineChangedForm: View {
+    // Properties to hold user input
+    var coords: CLLocationCoordinate2D
+    var pinDataStored: Artwork
+    
+    let statusDict = [0: "active", 1: "out_of_order", 2: "retired"]
+    @State private var name: String = ""
+    @State private var address: String = ""
+    @State private var area: String = ""
+    @State private var paywall: Bool = false
+    @State private var multimachine: String = ""
+    @State private var selectedSegment: Int = 0
+    @State private var lonLat: String = ""
+    
+    // to select the location on the map
+    @State private var showMap = false
+    @State private var selectedLocation: CLLocationCoordinate2D
+    @State private var isMapPresented = false
+    
+    @State private var showFinishedAlert = false
+    @State private var displayResponse: String = ""
+    @Environment(\.presentationMode) private var presentationMode // Access the presentationMode environment variable
+    @State private var showAlert = false
+    @State private var isLoading = false
+
+    @State private var keyboardHeight: CGFloat = 0
+    private var keyboardObserver: AnyCancellable?
+
+    init(pinData: Artwork) {
+        pinDataStored = pinData
+        _name = State(initialValue: pinData.title!)
+        coords = pinData.coordinate
+        _selectedLocation = State(initialValue: coords)
+        _address = State(initialValue: pinData.address)
+        _area = State(initialValue: pinData.area)
+        _paywall = State(initialValue: pinData.paywall)
+        _multimachine = State(initialValue: String(pinData.multimachine))
+        
+        let lonLatConverted = "\(coords.latitude)° N, \(coords.longitude)° O".replacingOccurrences(of: ".", with: ",")
+        _lonLat = State(initialValue: lonLatConverted)
+
+        switch pinData.machineStatus {
+            case "available":
+                _selectedSegment = State(initialValue: 0)
+            case "out-of-order":
+                _selectedSegment = State(initialValue: 1)
+            case "retired":
+                _selectedSegment = State(initialValue: 2)
+            default:
+                _selectedSegment = State(initialValue: 0)
+        }
+        // Observe keyboard frame changes
+        keyboardObserver = NotificationCenter.default.publisher(for: UIResponder.keyboardWillChangeFrameNotification)
+                .compactMap { $0.userInfo?["UIKeyboardFrameEndUserInfoKey"] as? CGRect }
+                .map { $0.height }
+                .subscribe(on: DispatchQueue.main)
+                .assign(to: \.keyboardHeight, on: self)
+    }
+    
+    var body: some View {
+        ScrollView{
+        VStack(alignment: .leading, spacing: 5) {
```

Comment:
The spacing is too gentle, I think we should reduce a bit the vertical space between the fields. Especially because we plan to add more fields to this edit form view.